### ISSUE TITLE
CDI scanning trace is very noisy

### ISSUE THREAD

**Issue by Azquelt** [OpenLiberty/open-liberty#28527]
When CDI is scanning a large application, the trace log is filled with many repetitive method calls with no useful information.

Two culprits I could immediately see were `BeanDeploymentArchiveImpl.hasBeenScanned` and `BeanDeploymentArchiveImpl.scanForBeanDefiningAnnotations`. Both of these are called recursively on each archive and all of its dependants.

This causes the trace logs to roll very quickly and important trace information to be lost when investigating service cases.

A first step would be to analyse both methods with `@Trivial` but we should really test deploying a CDI app with many archives which all need to be scanned (perhaps in the course of testing #28526) and analyse the trace produced to see where we're tracing useful information and where we're just making noise.

**Comment by benjamin-confino** [comment]
In my experience logging what's happening during CDI bean scanning is prime signal, over time I've been increasing rather than reducing the trace from that area. I don't think we'd want to make `scanForBeanDefiningAnnotations` trivial. 

`hasBeenScanned` could be trivial though. 

Another thing we could do is compress the trace from `getBeanDiscoveryMode` to one line, but we would have to check if nothing that happens between entering and exiting that method produces useful trace elsewhere. 

I also noticed. Why do we call `scanForBeanDefiningAnnotations` from WebSphereCDIDeploymentImpl? It looks like all that early call is doing is building a cache that will be read the next time `scanForBeanDefiningAnnotations` is called? We could skip that step and only populate the cache when we need it. That would skip a lot of logging and ensure the remaining logs from `scanForBeanDefiningAnnotations` are close to where the annotations are used to scan for beans which is where they'll be useful. 

I think we'd get a lot of this for free when we resolve #28526

**Comment by Azquelt** [comment]
I agree there's a lot of important info that we need to trace during bean scanning, however I would also say that we need to have a look at the amount of noise because it's causing trace logs to roll over too fast for large applications. If you don't capture the whole application startup the first time, it wastes time asking for another capture with more logs stored (aside from the annoyance of doing a large upload and analysing the large amount of logs).

I would suggest we look at this from two sides:
* Itemise the most commonly useful information and trace this at `event` level
  * I'd expect this would include the CDI config, the bean archives, their scan modes, the list of classes and which ones were detected as beans.
  * If this information is available to us while the server is running (rather than just at startup) we should probably also create a CDI dump introspector to include it in a server dump.
* Go through the full trace for a large application and identify which parts of the trace are just noise and can be removed or cut down (e.g. we don't want entry/exit trace for `hasBeenScanned`, but we might want the list of unscanned dependencies).
